SofaKernel/modules/SofaEngine/BoxROI.inl

```diff
@@ -517,6 +517,14 @@ void BoxROI<DataTypes>::doUpdate()
     if(!d_doUpdate.getValue()){
         return ;
     }
+    if (d_X0.getValue().size() == 0)
```

Comment:
thx for this bugfix. 
Could you change it to adopt the "one day wishes" mechanism of ComponentState check. 
I.e in the Init method where d_X0 is set, if not set/found put m_componentstate = ComponentState::Invalid ;
Then at start of the doUpdate method check if m_componentstate != ComponentState::Invalid ; otherwise return.

Reply:
Thanks for your review! Actually, I raised only a warning and did not set the component Invalid because d_X0 could be set later on, once all the components have been initialised, and the scene may then still work anyway.

Reply:
I thought it was a Data link. 
Normally isSet should return true even if the link depend on a component that will be init after.

Reply:
Yes indeed, isSet returns true but d_X0 remains empty, which is why there was a crash when doUpdate is looking for values inside d_X0. That is why I test if the value in d_X0 contains at least something.

Reply:
ok sorry for my missunderstood.